src/main/java/com/microsoft/graph/serializer/ISerializer.java

```diff
@@ -36,25 +37,25 @@
     /**
      * Deserialize an object from the input string
      * 
-     * @param inputString the string that stores the representation of the item
+     * @param inputStream the string that stores the representation of the item
      * @param clazz       the .class of the item to be deserialized
      * @param <T>         the type of the item to be deserialized
      * @return            the deserialized item from the input string
      */
     @Nullable
-    <T> T deserializeObject(@Nonnull final String inputString, @Nonnull final Class<T> clazz);
+    <T> T deserializeObject(@Nonnull final InputStream inputStream, @Nonnull final Class<T> clazz);
```

Comment:
can you add two overrides accepting strings please? That's because in a number of samples and snippets in the documentation we parse from a string value provided in the HTTP snippet. And having to get to an input stream first would complexify these samples.

Reply:
Sure, I can do that. Do you prefer me to use a default method in the interface, which will handle the override without headers? e.g.
```java
public interface ISerializer {
   default <T> T deserializeObject(@Nonnull final String inputString, @Nonnull final Class<T> clazz) {
       return deserializeObject(inputString, clazz, emptyMap())
   }
    
    <T> T deserializeObject(@Nonnull final String inputString, @Nonnull final Class<T> clazz, @Nonnull final Map<String, List<String>> responseHeaders);
}
```

Reply:
I didn't know Java had support for default interfaces. I like that approach, no need to force implementers to implement an overload just to call another one. Resolving